types/specification.d.ts

```diff
@@ -0,0 +1,105 @@
+/** Stamp Specification v1.6 */
+declare namespace Specification {
+  /** Base type for `properties` */
+  type PropertyMap = { [s: string]: any; };
+
+  /** Base type for `methods` */
+  type MethodMap = { [s: string]: (...args: Array<any>) => any; };
+
+  /** Extract the instance type from Factory, Descriptor (or Instance) */
+  type StampType<I> = I extends Factory<infer I> ? I
+                    : I extends Descriptor<infer I> ? I
+                    : I extends Instance ? I
+                    : never;
+
+  /** A composable object (`stamp` factory or Descriptor) */
+  type Composable<I, U = StampType<I>> = Factory<U> | Descriptor<U>;
+
+  /** Base `stamp` instance */
+  type Instance = { [s: string]: any; };
```

Comment:
This comment got be confused. I'd change "stamp instance" to "object instance", or "object instance created from stamp".